src/notes/notes.service.ts

```diff
@@ -43,147 +58,117 @@ export class NotesService {
     ];
   }
 
-  createNote(noteContent: string, alias?: NoteMetadataDto['alias']): NoteDto {
-    this.logger.warn('Using hardcoded data!');
-    return {
-      content: noteContent,
-      metdata: {
-        alias: alias,
-        createTime: new Date(),
-        description: 'Very descriptive text.',
-        editedBy: [],
-        id: 'foobar-barfoo',
-        permission: {
-          owner: {
-            displayName: 'foo',
-            userName: 'fooUser',
-            email: 'foo@example.com',
-            photo: '',
-          },
-          sharedToUsers: [],
-          sharedToGroups: [],
-        },
-        tags: [],
-        title: 'Title!',
-        updateTime: new Date(),
-        updateUser: {
-          displayName: 'foo',
-          userName: 'fooUser',
-          email: 'foo@example.com',
-          photo: '',
-        },
-        viewCount: 42,
-      },
-      editedByAtPosition: [],
-    };
+  async createNoteDto(
+    noteContent: string,
+    alias?: NoteMetadataDto['alias'],
+    owner?: User,
+  ): Promise<NoteDto> {
+    const note = await this.createNote(noteContent, alias, owner);
+    return this.toNoteDto(note);
   }
 
-  getNoteByIdOrAlias(noteIdOrAlias: string) {
-    this.logger.warn('Using hardcoded data!');
-    return {
-      content: 'noteContent',
-      metdata: {
-        alias: null,
-        createTime: new Date(),
-        description: 'Very descriptive text.',
-        editedBy: [],
-        id: noteIdOrAlias,
-        permission: {
-          owner: {
-            displayName: 'foo',
-            userName: 'fooUser',
-            email: 'foo@example.com',
-            photo: '',
-          },
-          sharedToUsers: [],
-          sharedToGroups: [],
-        },
-        tags: [],
-        title: 'Title!',
-        updateTime: new Date(),
-        updateUser: {
-          displayName: 'foo',
-          userName: 'fooUser',
-          email: 'foo@example.com',
-          photo: '',
-        },
-        viewCount: 42,
-      },
-      editedByAtPosition: [],
-    };
+  async createNote(
+    noteContent: string,
+    alias?: NoteMetadataDto['alias'],
+    owner?: User,
+  ): Promise<Note> {
+    const newNote = Note.create();
+    newNote.revisions = Promise.resolve([
+      Revision.create(noteContent, noteContent),
```

Comment:
Shouldn't it use the patch instead of noteContent or maybe use the revisionRepo?

Reply:
Yeah, that line is missing a `TODO`.